File: solutions/cpp/log-levels/1/log_levels.cpp
#include <string>

namespace log_line {
    std::string message(std::string line) {
        size_t delimiter_index{line.find(" ")};
        return line.substr(delimiter_index + 1);
    }

    std::string log_level(std::string line) {
        size_t end_index{line.find("]")};
        return line.substr(1, end_index - 1);
    }

    std::string reformat(std::string line) {
        return log_line::message(line) + " (" + log_line::log_level(line) + ")";
    }
}

File: solutions/java/log-levels/2/src/main/java/LogLevels.java
public class LogLevels {
    
    public static String message(String logLine) {
        int semicolonIndex = logLine.indexOf(":");
        return logLine.substring(semicolonIndex + 1).trim();
    }

    public static String logLevel(String logLine) {
        int semicolonIndex = logLine.indexOf(":");
        return logLine.substring(1, semicolonIndex - 1).toLowerCase();
    }

    public static String reformat(String logLine) {
        return String.format("%s (%s)", message(logLine), logLevel(logLine));
    }
}
